### DISCUSSION TITLE
How will the camouflage system work?

### DISCUSSION THREAD

**Discussion by carlssonemil** [carlssonemil/interstellar#1] [Ideas]
Hello fellow gamers and developers! 🎮 

As many of you are aware, the highly anticipated Call of Duty Modern Warfare 3 is set to be released in the coming months. One of the exciting features of the Call of Duty series has always been the vast array of camouflages that players can earn and apply to their weapons.

I'm starting this discussion thread to create a collaborative space where we can gather, organize, and track information about the different camouflages in MW3 as it becomes available. This will not only help players strategize on how to earn them but also assist in the data collection for the new tracker I'm creating.

Here's how you can help:

- **Information Sharing:** If you come across any official (or credible) source mentioning details about camouflages, drop the link and a brief summary here.
- **Design Ideas:** If you have ideas on how the tracker should look or function, sketch it out or describe it! It will most likely be very similar to the previous trackers I've made!
- **Collaboration:** If you're interested in helping out with the code, feel free to check out the [contributing guidelines](https://github.com/carlssonemil/mw3/blob/main/CONTRIBUTING.md).
- **Feedback:** Any constructive feedback on what we gather is always welcome.

Remember, we're looking for accurate and credible information. Please refrain from sharing rumors or unverified details.

Happy gaming! 🕹️

**Comment by carlssonemil** [comment]
From my understanding, and from [this blog post](https://www.callofduty.com/blog/2023/08/call-of-duty-modern-warfare-II-to-modern-warfare-III-carry-forward), all camouflages and weapons from CoD MW2 will carry over to MW3. Do we want to be able to track all the previous weapons and camouflages in the new tracker, or only the new ones? 🤔

**Reply by jamesallnutt** [reply]
Hey @carlssonemil - I was thinking about this - I think as MW2 will still be functional as it's own game, (Multiplayer / DMZ etc) - perhaps instead we add a quick "toggle" button to take the user between the MW2 site and the MW3 site. We could also add some further branding to differentiate between them clearly?

**Reply by carlssonemil** [reply]
It seems that MW3 is sort of like a DLC for MW2, so we might not need to create a new tracker at all. Maybe just adding the new camouflages to the Orion tracker site and then extend the filters so the users can filter between MW2 and MW3 will be enough 🤷 I'll wait for the game to be released and then give it a few days before making a decision on how we're moving forward 😄

**Reply by jamesallnutt** [reply]
Maybe worth rebranding from Orion to something else, as the mastery camos seem to have different names?

**Reply by carlssonemil** [reply]
Definitely!

**Reply by jamesallnutt** [reply]
@carlssonemil - here we go, the mastery camos!

![4 Interstellar](https://github.com/carlssonemil/mw3/assets/41279064/e157f4f5-ca4d-4033-a831-6882431bb69b)
![3 Priceless](https://github.com/carlssonemil/mw3/assets/41279064/c2483b4a-1e6e-4ba9-a0fb-5b15be304713)
![2 Forged](https://github.com/carlssonemil/mw3/assets/41279064/e02708d5-4b41-4c83-a06b-33c6a0be4d02)
![1 Gold](https://github.com/carlssonemil/mw3/assets/41279064/93066723-5cf7-4dc7-8dff-6ea70d05da51)

We could start the prep work now, we have the list of guns, if we base the assumption, 4 challenges per gun, then the masteries. We could prep the actual code, and then update with the actual challenges on release?

**Reply by carlssonemil** [reply]
Nice! 🚀 I'll get started tomorrow after work!

**Reply by jamesallnutt** [reply]
@carlssonemil I'm sorry to say there are even more... https://www.callofduty.com/blog/2023/11/call-of-duty-modern-warfare-III-launch-comms-challenges-weapon-camos

So not only are these the new mastery camos for MWIII multiplayer, we've also got another set of 4 for Zombies. However, it's 4 mastery camos for MWII guns for zombies, and a separate 4 camos for MWIII guns on zombies:

![MWII Zombies](https://github.com/carlssonemil/mw3/assets/41279064/3e391603-de5b-4b01-89ef-aa037019910f)

![MWIII Zombies](https://github.com/carlssonemil/mw3/assets/41279064/23395e45-8cf2-44ee-a606-4e1add6bc8ad)

We are also going to get weekly camo challenges via Weekly Challenges 😢 might have our updating work cut out for us - maybe worth setting up the Discord so we can have a convo on how we can structure / distribute the updates that are going to be needed?

**Reply by Pausbe** [reply]
It seems like this "carry forward" will apply to COD 2024 and future CODs as well, if we want to think far out.
![image](https://github.com/carlssonemil/interstellar/assets/74504687/dac88535-e2aa-4d75-a706-2b46a92c11e6) 
[source](https://www.callofduty.com/blog/2023/11/call-of-duty-modern-warfare-iii-warzone-hq-game-launcher-intel)

**Reply by jamesallnutt** [reply]
Yeah @Pausbe - so my understanding is, with the new Engine they adopted in MW2 - all future games will run on this same engine to speed up development, and features like Carry Forward. I imagine long term @carlssonemil will be open to ideas, otherwise its going to be so many filters haha

**Reply by carlssonemil** [reply]
I've been thinking of merging all the trackers into one tracker (with separate sub-pages for each game) since Vanguard was released. So it's probably a project I can look into until the next game. The new tracker "Interstellar" will probably evolve into a collection of all the trackers, and eventually rebranded, to more easily contain all previous trackers and any new coming in the future!

**Comment by whyyn0tt** [comment]
I'm also interested in helping this year. I've gathered screenshots and
reported bugs in years past, but would love to contribute more!
I don't have much expertise in development, but I work in data
scrubbing/analysis, and I have plenty of experience with software project
management and UAT...if this project warrants it.

Keep me in the loop and godspeed gents!

**Reply by jamesallnutt** [reply]
Data Scrubbing is a must 😂  @whyyn0tt especially for all the calling card challenges - so I imagine it would be welcomed!

**Comment by carlssonemil** [comment]
I've decided to create a new tracker and copy all content from the Orion tracker over to the new one. Makes it easier to keep them both separated and it gives me some room for improvements without having to refactor the existing code structure. It most likely won't be ready until after the weekend, but I think that's fine since it will give us some time to collect more information on the new camouflages and their requirements.

Probably won't have time for a Discord call though this weekend as I'm a bit busy, but I've created a server where we can easier discuss upcoming changes or seasons. I'll link it here tomorrow! I'll also open up the server for the community so it's easier to request new features and report bugs! 😄

**Reply by jamesallnutt** [reply]
That sounds great @carlssonemil - I'll be picking up my copy tomorrow, but can start work on creating a spreadsheet with all the camo challenges in it and share that on the discord or here. 

Would be good to discuss what your vision is for it, so we can make sure we are helping out as best we can

**Reply by whyyn0tt** [reply]
Insomnia got the better of me tonight, and I churned out this rough data capture workbook (2 tabs). I can fairly easily update this once we have the full list from the game.  The images are mostly just crude, quick edits from the blog post to make it pretty (lol).

Feel free to steal the data, or request access if you want to collab via docs;

[mw3-camo-data-capture](https://docs.google.com/spreadsheets/d/1wNyj3ik9znRLS6z8FYAvUCohiyty5RObg-Te42CIEMk/edit?usp=sharing)

**Reply by carlssonemil** [reply]
Here's the very basic [Discord server](https://discord.gg/gw9pbdhZ) I created. I'll spend some time improving this before adding a link in the tracker. I'll set your roles as "Developer" when you join so we can have a private text channel to discuss in!

A brief outline of my vision would be that it would function as the previous trackers basically, but we simply add a filter so that MW2 and MW3 guns/camos can be shown/hidden. I'll also add a separate page for the zombie camouflages. We'll wait and see how the weekly camo challenges will function before I make a decision on how we will show those in the tracker! I don't think I need any help with the code right now (as much of it is just copy-pasted from the Orion tracker with some slight improvements and modifications), but I do need help to collect all the camouflage name, requirements and images! 😄

**Reply by carlssonemil** [reply]
I'll also add import functionality from the Orion tracker so that the users don't need to input their progress from MW2 manually!

**Reply by jamesallnutt** [reply]
Sounds great @carlssonemil ! I’ll do my best to share any sheets I put together in the discord + json files as well I’ll do a pull request for

**Comment by whyyn0tt** [comment]
Oh baby, spreadsheets. See now you're talking my language! 😆

**Comment by nrobinson12** [comment]
Hey, I wouldn't mind helping out here and there. I'm new to open source and not sure if you need a hand. I have experience in front end with React and have been working as a Software Developer for a few years.

**Reply by carlssonemil** [reply]
Help is always welcomed! Right now I'm mostly just setting up the tracker similar to how the previous trackers works while also making some refactoring and improvements to the code base. Should be done with that after the weekend. Then I just need to add all the new camouflages and their requirements. I'll probably setup a base for the code and then create issues to fill it out if more people want to help, so keep an eye out on the issues tab. If I'm extremely bored next week I might just input them all myself though 😅

**Reply by carlssonemil** [reply]
Most of the work has already been done, but I didn't want to do too much before the game was released in case something changed in the grind and it would be wasted time 😄
